specs/market-api.yaml

```diff
@@ -549,131 +511,127 @@ paths:
+          $ref: 'common.yaml#/responses/Gone' # Expired
         default:
-          $ref: '#/components/responses/UnexpectedError'
-
+          $ref: 'common.yaml#/responses/UnexpectedError'
 
   /agreements/{agreementId}/confirm:
     parameters:
       - $ref: '#/components/parameters/agreementId'
+      - $ref: 'common.yaml#/parameters/appSessionId'
     post:
       tags:
         - requestor
       operationId: confirmAgreement
-      summary: Sends Agreement proposal to the Provider.
+      summary: ConfirmAgreement - Sends Agreement proposal to the Provider.
       description: >
         Signs self-created Agreement and sends it to the Provider.
 
         This call should immediately follow `createAgreement`.
       responses:
-        '204':
+        204:
           description: Agreement confirmed.
-        '401':
-          $ref: '#/components/responses/Unauthorized'
-        '404':
-          $ref: '#/components/responses/NotFound'
-        '409':
+        401:
+          $ref: 'common.yaml#/responses/Unauthorized'
+        404:
+          $ref: 'common.yaml#/responses/NotFound'
+        409:
           description: Agreement not in `Proposal` state.
-        '410':
-          $ref: '#/components/responses/Expired'
+        410:
+          $ref: 'common.yaml#/responses/Gone' # Expired
         default:
-          $ref: '#/components/responses/UnexpectedError'
+          $ref: 'common.yaml#/responses/UnexpectedError'
 
-
-  /agreements/{agreementId}/wait:
+  /agreements/events:
     parameters:
-      - $ref: '#/components/parameters/agreementId'
-      - $ref: 'common.yaml#/parameters/timeout'
-    post:
+      - $ref: 'common.yaml#/parameters/pollTimeout'
+      - $ref: 'common.yaml#/parameters/maxEvents'
+      - $ref: 'common.yaml#/parameters/appSessionId'
+    get:
       tags:
         - requestor
-      operationId: waitForApproval
-      summary: Waits for Agreement approval by the Provider.
+        - provider
+      operationId: collectAgreementEvents
+      summary: CollectAgreementEvents - Collects events related to an Agreement.
       description: >
-        This is a blocking operation. The call may be aborted by Requestor caller
-        code. After the call is aborted, another `waitForApproval` call can be
-        raised on the same Agreement Id.
+        This is a blocking operation.
 
-        It returns one of the following options:
+        It returns events such as:
+
+        * `AgreementProposalEvent` - Noew Agreement has been proposed by a Requestor.
 
-        * `Approved` - Indicates that the Agreement has been approved by the Provider.
+        * `AgreementApprovedEvent` - Indicates that the Agreement has been approved by the Provider.
           - The Provider is now ready to accept a request to start an Activity
             as described in the negotiated agreement.
           - The Providers’s corresponding `approveAgreement` call returns `Approved` after
             this on the Provider side.
```

Comment:
This needs rephrasing... and maybe even reconsideration, because it seems that we need to notify Provider when Requestor will receive this specific type of event (ie. `AgreementApprovedEvent`) to fulfill requirement which is stated here